src/gluonts/model/tpp/rmtpp/_estimator.py

```diff
@@ -0,0 +1,205 @@
+        assert (
+            num_training_instances > 0
+        ), "The value of `num_training_instances` should be > 0"
+
+        self.num_hidden_dimensions = num_hidden_dimensions
+        self.prediction_interval_length = prediction_interval_length
+        self.context_interval_length = (
+            context_interval_length
+            if context_interval_length is not None
+            else prediction_interval_length
+        )
+        self.num_marks = num_marks
+        self.embedding_dim = embedding_dim
+        self.num_parallel_samples = num_parallel_samples
+        self.num_training_instances = num_training_instances
+        self.freq = freq
+
+    def create_training_network(self) -> HybridBlock:
+        return RMTPPTrainingNetwork(
+            num_marks=self.num_marks,
+            interval_length=self.prediction_interval_length,
+            embedding_dim=self.embedding_dim,
+            num_hidden_dimensions=self.num_hidden_dimensions,
+        )
+
+    def create_transformation(self) -> Transformation:
+        return Chain(
+            [
+                ContinuousTimeInstanceSplitter(
+                    past_interval_length=self.context_interval_length,
+                    future_interval_length=self.prediction_interval_length,
+                    train_sampler=ContinuousTimeUniformSampler(
+                        num_instances=self.num_training_instances
+                    ),
+                ),
+                RenameFields(
+                    {
+                        "past_target": "target",
+                        "past_valid_length": "valid_length",
+                    }
+                ),
+            ]
+        )
+
+    def create_predictor(
+        self, transformation: Transformation, trained_network: HybridBlock
+    ) -> Predictor:
+        # TODO: Can we remove the comments?
+        # trained_params = trained_network.collect_params()
+        #
+        # del trained_network.collect_params().get("decay_bias").init
+
+        prediction_network = RMTPPPredictionNetwork(
+            num_marks=self.num_marks,
+            prediction_interval_length=self.prediction_interval_length,
+            interval_length=self.context_interval_length,
+            embedding_dim=self.embedding_dim,
+            num_hidden_dimensions=self.num_hidden_dimensions,
+            params=trained_network.collect_params(),
+            num_parallel_samples=self.num_parallel_samples,
+        )
+
+        return PointProcessGluonPredictor(
+            input_names=["target", "valid_length"],
+            prediction_net=prediction_network,
+            batch_size=self.trainer.batch_size,
+            prediction_interval_length=self.prediction_interval_length,
+            freq=self.freq,
+            ctx=self.trainer.ctx,
+            input_transform=transformation,
+        )
+
+    # TODO: Mypy complains about invalid signature
+    @no_type_check
+    def train_model(
```

Comment:
There seems to be a missing argument which was probably introduced recently: `shuffle_buffer_length`
```
    def train_model(
        self,
        training_data: Dataset,
        validation_data: Optional[Dataset] = None,
        num_workers: Optional[int] = None,
        num_prefetch: Optional[int] = None,
        shuffle_buffer_length: Optional[int] = None,
        **kwargs,
    ) -> TrainOutput:
        return super().train_model(
            training_data,
            validation_data,
            num_workers,
            num_prefetch,
            shuffle_buffer_length,
            batchify_fn=partial(batchify, variable_length=True),
            **kwargs,
        )
```